azure-pipelines/official.yml

```diff
@@ -42,24 +42,33 @@ parameters:
   type: boolean
   default: true
 
-stages:
+resources:
+  repositories:
+  - repository: MicroBuildTemplate
+    type: git
+    name: 1ESPipelineTemplates/MicroBuildTemplate
+    ref: refs/tags/release
 
-- stage: Build
-  variables:
-    DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
-    BuildConfiguration: Release
-    NUGET_PACKAGES: $(Agent.TempDirectory)/.nuget/packages
-    SignTypeSelection: ${{ parameters.SignTypeSelection }}
-    Packaging.EnableSBOMSigning: false
-    Codeql.Enabled: true
-
-  jobs:
-  - template: build.yml
-    parameters:
-      EnableCompliance: ${{ parameters.EnableCompliance }}
-      EnableAPIScan: ${{ parameters.EnableAPIScan }}
-      windowsPool: VSEngSS-MicroBuild2022-1ES
-      includeMacOS: ${{ parameters.includeMacOS }}
-      RunTests: ${{ parameters.RunTests }}
-
-- template: prepare-insertion-stages.yml
+extends:
+  template: azure-pipelines/MicroBuild.1ES.Official.yml@MicroBuildTemplate
+  parameters:
+    sdl:
+      sourceAnalysisPool: VSEngSS-MicroBuild2022-1ES
+    stages:
+    - stage: Build
+      variables:
+        DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+        BuildConfiguration: Release
+        NUGET_PACKAGES: $(Agent.TempDirectory)/.nuget/packages
+        SignTypeSelection: ${{ parameters.SignTypeSelection }}
+        Packaging.EnableSBOMSigning: false
+        Codeql.Enabled: true
+      jobs:
+      - template: /azure-pipelines/build.yml@self
+        parameters:
+          EnableCompliance: ${{ parameters.EnableCompliance }}
```

Comment:
I wonder if this `EnableCompliance` switch should be tied to whether we use the 'official' or 'unofficial' variety of the templates.

Reply:
> We are unable to use a variable for this.

I agree. But this is not a _variable_. As a _parameter_, it has much more ability to influence the structure of YAML, and in particular I think you should be able to use something like this:

```yaml
${{ if parameters.EnableCompliance }}:
  extends:
    template: azure-pipelines/MicroBuild.1ES.Official.yml@MicroBuildTemplate
    ....
${{ else }}:
  extends:
    template: azure-pipelines/MicroBuild.1ES.Unofficial.yml@MicroBuildTemplate
    ....
```

Reply:
ohhhh.. Let's give it a whirl!  I don't know yaml well enough....Argh!

Reply:
@AArnott , so this is weird.  I'm having trouble getting this to work.  And, when I say cannot get it to work, if the EnableCompliance is set to true, it works.  But, if I toggle the EnableCompliance to False, it completely ignores the template line and does not extend the template at all.

I've tried the following:
1. Adding a "-" at the beginning of the if/else statements.
2. I've changed the if else statement to use syntax like this instead:
  ${{ eq(parameters.EnableCompliance, 'true') }}:
and 
  ${{ ne(parameters.EnableCompliance, 'true') }}:
3. I've tried moving the if/else statements inside of the extends: like this:
extends::
  {{ if ... }}:
    template:
  {{ else ...}}:
    template:

Nothing so far is working for it to switch between the two different templates.  Quite bizarre.
Any thoughts as to why this might be the case?  Thx.

Reply:
I pushed to your branch a change that seems to work. At least, it parses. The unofficial branch actually starts to build. The official one complains that the user steps include a `publish` step.
And I had to comment out the `prepare-insertion.steps.yml` template reference to avoid a queue-time failure where `stages` is unexpected.

Reply:
@AArnott , thanks!  I must have done something wrong because I couldn't get the else part to work at all. 
I'm hoping the prepare-insertion.steps.yml is just a spacing issue.  
Try again, please?

Reply:
I tried queuing. With compliance enabled, the error was:

> 
/v1/Steps/UserProvidedSteps.yml@1esPipelines (Line: 45, Col: 7): Unexpected value 'publish alias is not allowed. Please use `output: pipelineArtifact` instead. See https://eng.ms/docs/cloud-ai-platform/devdiv/one-engineering-system-1es/1es-docs/1es-pipeline-templates/features/outputs for more details'

Without compliance, [this is the run](https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=8963582&view=results).